lib/kredis/types/proxy/failsafe.rb

```diff
@@ -1,16 +1,21 @@
 module Kredis::Types::Proxy::Failsafe
-  def initialize(*)
-    super
-    @fail_safe_suppressed = false
+  extend ActiveSupport::Concern
+
+  included do
+    mattr_accessor :fail_safe_enabled, default: true
   end
 
-  def failsafe
-    yield
+  def failsafe(returning: nil, &block)
```

Comment:
What about changing `failsafe(returning: nil, &block)` API to failsafe(returning: nil, errors: [Redis::BaseError], &block) such that we make explicit which errors we catch?